transport.go

```diff
@@ -145,6 +120,14 @@ func (t *transport) Dial(ctx context.Context, raddr ma.Multiaddr, p peer.ID) (tp
 	if err != nil {
 		return nil, err
 	}
+
+	// Should be ready by this point, don't block.
+	var remotePubKey ic.PubKey
+	select {
+	case remotePubKey = <-keyCh:
+	default:
```

Comment:
We should return an error in this case.

Reply:
That applies to [the equivalent code in go-libp2p-tls](https://github.com/libp2p/go-libp2p-tls/blob/cc53c4f8deb88e5de59405fb70ef27c93ba57ae1/transport.go#L102-L107) as well. I'll prepare a PR.

Reply:
Same as in go-libp2p-tls, we probably also need to check if the channel is closed. Also, this check is backwards (return in the default case).